relayer/crates/starknet-chain-components/src/impls/messages/create_client.rs

```diff
@@ -73,10 +74,21 @@ where
             chain_id: payload.client_state.chain_id,
         };
 
+        // Convert Vec<u8> to a slice of 8 u32 values.
+        let mut root_u32: Vec<u32> = Vec::new();
+        for chunk in root.chunks_exact(4) {
```

Comment:
The root should be converted from `Vec<u8>` to `[u8; 32]` first using `TryFrom`, before converting again to `[u32; 8]`. With that, we can also use simpler loops like `for i in 0..8` and access the fixed size array directly. 

Also, we should move this logic to a dedicated function, and include documentation about how the conversion is done. i.e. using little endian to pack 4 `u8` into `u32`.

Reply:
Done in ee085c68f267ab8c7fce9b099ef4da7084a9fd64